database/src/sanitizer.py

```diff
@@ -89,3 +107,126 @@ def generate_embedding_tags(self, row):
         ]
 
         return ", ".join(text_fields).lower()
+
+    def clean_json_string(self, json_str):
+        # Clean and format JSON string for parsing
+        if pd.isna(json_str):
+            return "[]"
+
+        try:
+            if isinstance(json_str, str):
+                # Remove any leading/trailing whitespace
+                cleaned = json_str.strip()
+
+                # Replace single quotes with double quotes
+                cleaned = cleaned.replace("'", '"')
+
+                # Replace boolean values
+                cleaned = cleaned.replace("True", "true").replace("False", "false")
+
+                # Handle multiple JSON objects separated by space
+                if "} {" in cleaned:
+                    parts = cleaned.split("} {")
+                    # Take only the first object (color attributes)
+                    cleaned = parts[0] + "}"
+
+                # Handle array-like strings
+                if cleaned.startswith("[{") and cleaned.endswith("}]"):
+                    # Already in correct format
+                    pass
+                elif cleaned.startswith("{") and cleaned.endswith("}"):
+                    # Single object, wrap in array
+                    cleaned = f"[{cleaned}]"
+                else:
+                    # Invalid format, return empty array
+                    return "[]"
+
+                # Validate JSON structure
+                json.loads(cleaned)  # This will raise JSONDecodeError if invalid
+                return cleaned
+            else:
+                # If input is not a string, convert it to JSON string
+                return json.dumps(
+                    [json_str] if not isinstance(json_str, list) else json_str
+                )
+        except json.JSONDecodeError as e:
+            print(f"Error cleaning JSON string: {e}")
+            print(f"Original string: {json_str}")
+            return "[]"
+
+    def separate_colors(self, row):
+        try:
+            # Parse variation attributes to get color options
+            variation_data = json.loads(
+                self.clean_json_string(row["variationAttributes"])
+            )
```

Comment:
with changes to `clean_array_string` you should be able to access the data directly via row["variationAttributes"]